src/main/java/org/jabref/gui/importer/fetcher/WebSearchPaneViewModel.java

```diff
@@ -94,10 +95,22 @@ public void search() {
         BackgroundTask<ParserResult> task = BackgroundTask.wrap(() -> new ParserResult(activeFetcher.performSearch(getQuery().trim())))
                                                           .withInitialMessage(Localization.lang("Processing %0", getQuery()));
 
-        task.onFailure(dialogService::showErrorDialogAndWait);
+        task.onFailure(this::exceptionHandler);
 
         ImportEntriesDialog dialog = new ImportEntriesDialog(frame.getCurrentBasePanel().getBibDatabaseContext(), task);
         dialog.setTitle(activeFetcher.getName());
         dialog.showAndWait();
     }
+
+    private void exceptionHandler(Exception exception) {
+        if (exception instanceof FetcherException) {
+            if (exception.getMessage().equals("A network error occurred")) {
+                dialogService.showWarningDialogAndWait(Localization.lang("An error occurred"),
+                        Localization.lang(exception.getMessage() + ". You have no rights to access resources."));
+            }
+        } else {
+            dialogService.showWarningDialogAndWait(Localization.lang("An error occurred"),
+                    Localization.lang(exception.getMessage()));
```

Comment:
`Localization.lang` only works for static messages so sadly cannot be used here.

Reply:
Background information is available at https://devdocs.jabref.org/getting-into-the-code/code-howtos#using-localization-correctly.